stock_lot.py

```diff
@@ -83,28 +76,24 @@ def pack(cls, shipments):
     def lots_for_quality_tests(self):
         return list(set(m.lot for m in self.outgoing_moves if m.lot and
             m.state == 'draft' and
-            m.product.template.shipment_out_quality_template))
-
-
-class ShipmentInternal(CreateQualityLotTestsMixin, metaclass=PoolMeta):
-    __name__ = 'stock.shipment.internal'
-
-    @classmethod
-    def assign(cls, shipments):
-        super().assign(shipments)
-        cls.create_lot_quality_tests(shipments, 'shipment_internal')
-
-    def lots_for_quality_tests(self):
-        return list(set(m.lot for m in self.moves if m.lot and
-            m.state == 'assigned' and
-            m.product.template.shipment_internal_quality_template))
+            m.product.template.shipment_out_quality_template and
+            not [x for x in m.lot.quality_tests if
+            m.product.template.shipment_out_quality_template in x]))
 
 
 class Lot(metaclass=PoolMeta):
     __name__ = 'stock.lot'
     quality_tests = fields.One2Many('quality.test', 'document', 'Tests',
         readonly=True)
 
+    @classmethod
+    def copy(cls, lots, default=None):
+        if default is None:
+            default = {}
+        default = default.copy()
+        default['quality_test'] = None
```

Comment:
Should be "quality_tests", not "quality_test"

Reply:
done